src/test/java/api/support/fixtures/RecordCreator.java

```diff
@@ -1,53 +1,32 @@
 package api.support.fixtures;
 
-import java.util.HashMap;
-import java.util.HashSet;
-import java.util.Map;
-import java.util.Objects;
-import java.util.Optional;
-import java.util.Set;
-import java.util.UUID;
 import java.util.function.Function;
 
 import org.folio.circulation.support.http.client.IndividualResource;
+import org.folio.circulation.support.http.client.Response;
 
 import api.support.builders.Builder;
 import api.support.http.ResourceClient;
 import io.vertx.core.json.JsonObject;
 
-//TODO May need to support checking if record exists in store already?
 class RecordCreator {
   private final ResourceClient client;
-  private final Map<String, IndividualResource> identityMap;
-  private final Set<UUID> createdRecordIds;
   private final Function<JsonObject, String> identityMapKey;
 
   RecordCreator(
     ResourceClient client,
     Function<JsonObject, String> identityMapKey) {
 
     this.client = client;
-    this.identityMap = new HashMap<>();
-    this.createdRecordIds = new HashSet<>();
     this.identityMapKey = identityMapKey;
   }
 
   private IndividualResource create(JsonObject record) {
-
-    final IndividualResource createdRecord = client.create(record);
-
-    createdRecordIds.add(createdRecord.getId());
-
-    return createdRecord;
```

Comment:
Why has the identity map been removed?

Reply:
It was replaced with reading all records and matching by the key value instead of storing map internally. This allowed to get rid of map sync issues (e.g. I need to create ServicePointFixture in place instead of passing it as constructor argument to FeeFineOwnerFixture class).

Reply:
Thank you for explaining the reasoning. Does that mean the primary reason for this was to try to avoid dependencies between the fixtures?

My concern with removing this is that it means that every record creation now requires an additional HTTP request to check if the record already exists or not, which will slow down the tests.

Reply:
> Thank you for explaining the reasoning. Does that mean the primary reason for this was to try to avoid dependencies between the fixtures?

Yes.

> My concern with removing this is that it means that every record creation now requires an additional HTTP request to check if the record already exists or not, which will slow down the tests.

I've improved a bit the RecordCreator class. It now checks if the record present in map, if no - executes HTTP call.

Reply:
Ok, thank you, I think that is a good improvement